lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -158,21 +155,39 @@ class QcFlagService {
      * @return {Promise<void>} resolve once all periods are updated
      */
     async _updateEffectivePeriods({ from: newerPeriodFrom, to: newerPeriodTo }, periods) {
+        /**
+         * When run's timestamps are missing, effective periods have null timestamps, therefore
+         * comparison must be evaluated on extended set of values
+         * so  `from` == null is extended to -1, which is interpreted as run.startTime
+         * and `to`   == null is extended to MAX_DATE, which is interpreted as run.endTime
+         */
+        const MAX_DATE = new Date('100000-01-01').getTime();
+        const newerFromEval = newerPeriodFrom ?? -1;
+        const newerToEval = newerPeriodTo ?? MAX_DATE;
```

Comment:
This kind of construction is to avoid, it makes the code very hard to grasp, often for not much benefit. It's better to simply handle the null case in your if, extracting variable if needed to make things clear.

Reply:
refactored